Editor/Windows/FolderPanel.cs

```diff
@@ -152,6 +161,32 @@ public void EnsureFolderExists()
             CreateNewFolder<PaletteFolder>(null, InitialFolderName);
             window.CurrentCollectionSerializedObject.ApplyModifiedPropertiesWithoutUndo();
         }
+        
+        private void EnsureFoldersGuids()
+        {
+            for (int startIndex = 0; startIndex < window.FoldersSerializedProperty.arraySize; startIndex++)
+            {
+                SerializedProperty folderProperty = window.FoldersSerializedProperty.GetArrayElementAtIndex(startIndex);
+                EnsureFolderIDRecursively(folderProperty);
+            }
+
+            window.ApplyModifiedProperties();
+        }
+
+        private void EnsureFolderIDRecursively(SerializedProperty folderProperty)
+        {
+            if (!string.IsNullOrEmpty(folderProperty.FindPropertyRelative(GuidPropertyName).stringValue))
+                return;
+
+            folderProperty.FindPropertyRelative(GuidPropertyName).stringValue = Guid.NewGuid().ToString();
```

Comment:
Same question here: Moon was obsessed with not using strings for GUIDs right? In this particular case it does not need to be globally unique either. Is it worth making this something else, like an integer ticking up?

Reply:
The problem with using an integer that keeps ticking up is that when managing this integer like we can't use the array length for instance, because of children's folders. I didn't find a proper way of doing it, and also could be even harder to maintaining this between collections, to ensure unique IDs per folder. Yeah guids are less efficient than simple ints, but are more reliable in that sense.
We could reuse the LongGuid I created on SOC, that's like the best of both worlds.

Reply:
Also, I think the performance its incredibly important for runtime code, but for the editor side, unity already uses string-based guids everywhere, so I don't think it's that big of a deal.

Reply:
> The problem with using an integer that keeps ticking up is that when managing this integer like we can't use the array length for instance, because of children's folders. I didn't find a proper way of doing it

Isn't it a matter of just ticking up the integer whenever a new ID is requested? What do array lengths have to do with it 🤔 ?

> could be even harder to maintaining this between collections, to ensure unique IDs per folder

What I was getting at is that they don't _need_ to be unique between collections, right? They just need to be unique within the collection so that we can serialize which folder is currently selected. The Selected Folder is only for the current collection anyway, if you change collections your selected folder resets to the default folder.

Reply:
Gave it a go, seems to works well.